multisense_ros/src/multisense.cpp

```diff
@@ -0,0 +1,1651 @@
+
+//
+// Provide compiler with definition of the static members
+
+constexpr char MultiSense::LEFT[];
+constexpr char MultiSense::RIGHT[];
+constexpr char MultiSense::AUX[];
+constexpr char MultiSense::IMU[];
+
+constexpr char MultiSense::LEFT_CAMERA_FRAME[];
+constexpr char MultiSense::RIGHT_CAMERA_FRAME[];
+constexpr char MultiSense::LEFT_RECTIFIED_FRAME[];
+constexpr char MultiSense::RIGHT_RECTIFIED_FRAME[];
+constexpr char MultiSense::AUX_CAMERA_FRAME[];
+constexpr char MultiSense::AUX_RECTIFIED_FRAME[];
+constexpr char MultiSense::IMU_FRAME[];
+
+constexpr char MultiSense::INFO_TOPIC[];
+constexpr char MultiSense::STATUS_TOPIC[];
+constexpr char MultiSense::HISTOGRAM_TOPIC[];
+constexpr char MultiSense::RAW_CONFIG_TOPIC[];
+constexpr char MultiSense::MONO_TOPIC[];
+constexpr char MultiSense::RECT_TOPIC[];
+constexpr char MultiSense::DISPARITY_TOPIC[];
+constexpr char MultiSense::DISPARITY_IMAGE_TOPIC[];
+constexpr char MultiSense::DEPTH_TOPIC[];
+constexpr char MultiSense::OPENNI_DEPTH_TOPIC[];
+constexpr char MultiSense::COST_TOPIC[];
+constexpr char MultiSense::COLOR_TOPIC[];
+constexpr char MultiSense::RECT_COLOR_TOPIC[];
+constexpr char MultiSense::POINTCLOUD_TOPIC[];
+constexpr char MultiSense::LUMA_POINTCLOUD_TOPIC[];
+constexpr char MultiSense::COLOR_POINTCLOUD_TOPIC[];
+constexpr char MultiSense::ORGANIZED_POINTCLOUD_TOPIC[];
+constexpr char MultiSense::COLOR_ORGANIZED_POINTCLOUD_TOPIC[];
+constexpr char MultiSense::MONO_CAMERA_INFO_TOPIC[];
+constexpr char MultiSense::RECT_CAMERA_INFO_TOPIC[];
+constexpr char MultiSense::COLOR_CAMERA_INFO_TOPIC[];
+constexpr char MultiSense::RECT_COLOR_CAMERA_INFO_TOPIC[];
+constexpr char MultiSense::DEPTH_CAMERA_INFO_TOPIC[];
+constexpr char MultiSense::DISPARITY_CAMERA_INFO_TOPIC[];
+constexpr char MultiSense::COST_CAMERA_INFO_TOPIC[];
+constexpr char MultiSense::IMU_TOPIC[];
+
+MultiSense::MultiSense(const std::string& node_name,
+               const rclcpp::NodeOptions& options,
+               std::unique_ptr<multisense::Channel> channel,
+               const std::string& tf_prefix,
+               bool use_image_transport,
+               bool use_sensor_qos):
+    Node(node_name, options),
+    channel_(std::move(channel)),
+    left_node_(create_sub_node(LEFT)),
+    right_node_(create_sub_node(RIGHT)),
+    aux_node_(create_sub_node(AUX)),
+    imu_node_(create_sub_node(IMU)),
+    frame_id_left_(tf_prefix + LEFT_CAMERA_FRAME),
+    frame_id_right_(tf_prefix + RIGHT_CAMERA_FRAME),
+    frame_id_aux_(tf_prefix + AUX_CAMERA_FRAME),
+    frame_id_rectified_left_(tf_prefix + LEFT_RECTIFIED_FRAME),
+    frame_id_rectified_right_(tf_prefix + RIGHT_RECTIFIED_FRAME),
+    frame_id_rectified_aux_(tf_prefix + AUX_RECTIFIED_FRAME),
+    frame_id_imu_(tf_prefix + IMU_FRAME),
+    static_tf_broadcaster_(std::make_shared<tf2_ros::StaticTransformBroadcaster>(*this))
+{
+    using namespace std::chrono_literals;
+
+    //
+    // All streams off
+
+    stop();
+
+    info_ = channel_->get_info();
```

Comment:
didn't see yet if `Multisense` is expected to be instantiated by client code, but might we want a null check on channel then?